lib/Doctrine/ORM/Cache/DefaultQueryCache.php

```diff
@@ -260,7 +260,7 @@ public function put(QueryCacheKey $key, ResultSetMapping $rsm, $result, array $h
             throw new CacheException("Second-level cache query supports only select statements.");
         }
 
-        if (isset($hints[Query\SqlWalker::HINT_PARTIAL]) && $hints[Query\SqlWalker::HINT_PARTIAL]) {
+        if (($hints[Query\SqlWalker::HINT_PARTIAL] ?? false) === true || ($hints[Query::HINT_FORCE_PARTIAL_LOAD] ?? false) === true) {
```

Comment:
Then we can do this:

```suggestion
        if (($hints[Query\SqlWalker::HINT_PARTIAL] ?? $hints[Query::HINT_FORCE_PARTIAL_LOAD] ?? false) === true) {
```

Reply:
I like to keep it this way now.